File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        vector<pair<int, int>> numWithIndex;
        
        // Create a vector of pairs (number, index)
        for (int i = 0; i < n; i++) {
            numWithIndex.push_back({nums[i], i});
        }
        
        // Sort the vector of pairs
        sort(numWithIndex.begin(), numWithIndex.end());
        
        int left = 0;
        int right = n - 1;
        
        while (left < right) {
            int sum = numWithIndex[left].first + numWithIndex[right].first;
            
            if (sum == target) {
                return {numWithIndex[left].second, numWithIndex[right].second};
            } else if (sum < target) {
                left++;
            } else {
                right--;
            }
        }
        
        return {}; // No solution found
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mp = {}
        for i in range(len(nums)) :
            temp = target - nums[i]
            if temp in mp:
                return [i,mp[temp]]
            mp[nums[i]] = i 
        